packages/zoe/src/contracts/priceAggregator.js

```diff
@@ -0,0 +1,335 @@
+      await Promise.all(querierPs);
+    },
+  };
+  E(repeaterP).schedule(waker);
+
+  /**
+   * @param {Object} param0
+   * @param {number} [param0.overrideValueOut]
+   * @param {Timestamp} [param0.timestamp]
+   */
+  const makeCreateQuote = ({ overrideValueOut, timestamp } = {}) =>
+    /**
+     * @param {PriceQuery} priceQuery
+     * @returns {ERef<PriceQuote>=}
+     */
+    function createQuote(priceQuery) {
+      // Sniff the current baseValueOut.
+      const valueOutForBaseValueIn =
+        overrideValueOut === undefined
+          ? lastValueOutForBaseValueIn // Use the latest value.
+          : overrideValueOut; // Override the value.
+      if (valueOutForBaseValueIn === undefined) {
+        // We don't have a quote, so abort.
+        return undefined;
+      }
+
+      /**
+       * @param {Amount} amountIn the given amountIn
+       * @returns {Amount} the amountOut that will be received
+       */
+      const calcAmountOut = amountIn => {
+        const valueIn = mathIn.getValue(amountIn);
+        return mathOut.make(
+          floorDivide(multiply(valueIn, valueOutForBaseValueIn), baseValueIn),
+        );
+      };
+
+      /**
+       * @param {Amount} amountOut the wanted amountOut
+       * @returns {Amount} the amountIn needed to give
+       */
+      const calcAmountIn = amountOut => {
+        const valueOut = mathOut.getValue(amountOut);
+        return mathIn.make(
+          ceilDivide(multiply(valueOut, baseValueIn), valueOutForBaseValueIn),
+        );
+      };
+
+      // Calculate the quote.
+      const quote = priceQuery(calcAmountOut, calcAmountIn);
+      if (!quote) {
+        return undefined;
+      }
+
+      const {
+        amountIn,
+        amountOut,
+        timestamp: theirTimestamp = timestamp,
+      } = quote;
+      mathIn.coerce(amountIn);
+      mathOut.coerce(amountOut);
+      if (theirTimestamp !== undefined) {
+        return authenticateQuote([
+          { amountIn, amountOut, timer, timestamp: theirTimestamp },
+        ]);
+      }
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now =>
+          authenticateQuote([{ amountIn, amountOut, timer, timestamp: now }]),
+        );
+    };
+
+  /**
+   * @param {Array<number>} samples
+   * @param {Timestamp} timestamp
+   */
+  const updateQuote = async (samples, timestamp) => {
+    const median = calculateMedian(
+      samples.filter(sample => sample > 0 && Number.isSafeInteger(sample)),
+      { add, divide: floorDivide, isGTE },
+    );
+
+    // console.error('found median', median, 'of', samples);
+    if (median === undefined) {
+      return;
+    }
+
+    const amountOut = mathOut.make(median);
+
+    /** @type {PriceDescription} */
+    const quote = {
+      amountIn: baseAmountIn,
+      amountOut,
+      timer,
+      timestamp,
+    };
+
+    // Authenticate the quote by minting it with our quote issuer, then publish.
+    const authenticatedQuote = await authenticateQuote([quote]);
```

Comment:
Hm, we could potentially pass a promise for a quote. Why do we need to await?

Reply:
Because the `getPriceNotifier` is lossy.  We can fix that by deprecating it and introducing a new `getQuoteNotifier`, using @erights new lossy/lossless hybrid (such as was implemented for `map-unum.js`).

Reply:
Cool!